src/libraries/Native/Unix/System.Security.Cryptography.Native/openssl.c

```diff
@@ -1263,6 +1264,21 @@ static int32_t EnsureOpenSsl10Initialized()
     #define OPENSSL_INIT_NO_ATEXIT 0x00080000L
 #endif
 
+int volatile g_str_read_count = 0;
+int volatile g_err_unloaded = 0;
+
+static void HandleShutdown()
+{
+    // Set this first, which stops new callers from using the error string tables.
+    g_err_unloaded = 1;
+
+    // Now, spin until existing calls all finish, and we can move on with shutdown.
```

Comment:
Is it possible such calls could block or otherwise take an unbounded amount of time?

Reply:
Nah, they are just a TryGetValue and a snprintf.